kratos/sources/model_part.cpp

```diff
@@ -855,6 +855,293 @@ void ModelPart::RemoveElementsFromAllLevels(Flags identifier_flag)
+                                                                                    IndexType Id, 
+                                                                                    ModelPart::DofsVectorType& rMasterDofsVector,
+                                                                                    ModelPart::DofsVectorType& rSlaveDofsVector,
+                                                                                    const ModelPart::MatrixType& RelationMatrix,
+                                                                                    const ModelPart::VectorType& ConstantVector)
+{
+
+    KRATOS_TRY
+    if (IsSubModelPart())
+    {
+        MasterSlaveConstraintType::Pointer p_new_constraint = mpParentModelPart->CreateNewMasterSlaveConstraint(ConstraintName, Id, rMasterDofsVector,
+                                                                                                                    rSlaveDofsVector,
+                                                                                                                    RelationMatrix,
+                                                                                                                    ConstantVector);
+        mMasterSlaveConstraints.push_back(p_new_constraint);
+        mMasterSlaveConstraints.Unique();
+
+        return p_new_constraint;
+    }
+
+    auto existing_element_iterator = this->mMasterSlaveConstraints.find(Id);
+    if(existing_element_iterator != this->MasterSlaveConstraintsEnd() )
+        KRATOS_ERROR << "trying to construct an master-slave constraint with ID " << Id << " however a constraint with the same Id already exists";
+
+
+    //create the new element
+    MasterSlaveConstraintType const& r_clone_constraint = KratosComponents<MasterSlaveConstraintType>::Get(ConstraintName);
+    MasterSlaveConstraintType::Pointer p_new_constraint = r_clone_constraint.Create(Id, rMasterDofsVector,
+                                                                                        rSlaveDofsVector,
+                                                                                        RelationMatrix,
+                                                                                        ConstantVector);
+
+    mMasterSlaveConstraints.push_back(p_new_constraint);
+    mMasterSlaveConstraints.Unique();
+
+    return p_new_constraint;
+    KRATOS_CATCH("")
+
+}
+
+MasterSlaveConstraint::Pointer ModelPart::CreateNewMasterSlaveConstraint(const std::string& ConstraintName, 
+                                                                                    ModelPart::IndexType Id, 
+                                                                                    ModelPart::NodeType& rMasterNode,
+                                                                                    const ModelPart::DoubleVariableType& rMasterVariable,
```

Comment:
Are the Master and the Slavevariable Historical variables?

If so, then I would add a debug check if these variables are in the modelpart